extensions/parquet/table/src/main/java/io/deephaven/parquet/table/util/S3SeekableChannelProvider.java

```diff
@@ -0,0 +1,87 @@
+package io.deephaven.parquet.table.util;
+
+import io.deephaven.parquet.base.util.SeekableChannelsProvider;
+import io.deephaven.parquet.table.S3ParquetInstructions;
+import org.jetbrains.annotations.NotNull;
+import software.amazon.awssdk.http.async.SdkAsyncHttpClient;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.http.crt.AwsCrtAsyncHttpClient;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * {@link SeekableChannelsProvider} implementation that is used to fetch objects from AWS S3 instances.
+ */
+public final class S3SeekableChannelProvider implements SeekableChannelsProvider {
+
+    private final S3AsyncClient s3AsyncClient;
+    private final int fragmentSize;
+    private final int maxCacheSize;
+    private final int readAheadCount;
+    private final Duration readTimeout;
+    private final Map<Long, ChannelContext> contextMap = new HashMap<>(); // TODO Remove this
+
+    public S3SeekableChannelProvider(final S3ParquetInstructions s3Instructions) {
+        final String awsRegionName = s3Instructions.awsRegionName();
+        final int maxConcurrentRequests = s3Instructions.maxConcurrentRequests();
+        final SdkAsyncHttpClient asyncHttpClient = AwsCrtAsyncHttpClient.builder()
+                .maxConcurrency(maxConcurrentRequests)
+                .connectionTimeout(s3Instructions.connectionTimeout())
+                .build();
+        this.s3AsyncClient = S3AsyncClient.builder()
+                .region(Region.of(awsRegionName))
+                .httpClient(asyncHttpClient)
+                .build();
```

Comment:
I'm not sure how heavy-weight creating an S3AsyncClient is. It's possible that a caller might want to use the same S3 client on blobs that have different fragment sizes / read ahead, etc.

I want to make sure we don't over-expose public methods. I think this class / constructor could be made package private and moved to same level where it's needed?

Reply:
We do have configurable maxConcurrentRequests and connectionTimeout instructions being sent with each request that are used for constructing the async client. But your concern is very reasonable. I don't expect clients to change these configurations very frequently.
For now, I think we can go with the current design with a TODO because the performance bottleneck generally is in fetching the objects from aws rather than local operations.
What do you think?